fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -0,0 +1,215 @@
+        self.initial_daemon_status()
+
+        # ToDo: Verify that FAPD_LOGPATH works in the pkexec environment
+        if os.environ.get("FAPD_LOGPATH"):
+            self.fapd_profiling_stdout = os.environ.get("FAPD_LOGPATH") + ".stdout"
+            self.fapd_profiling_stderr = os.environ.get("FAPD_LOGPATH") + ".stderr"
+        else:
+            self.fapd_profiling_stdout = None
+            self.fapd_profiling_stderr = None
+
+    # ############### Public Interface ###############
+    def start_online_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        self.set_mode(FapdMode.ONLINE)
+        self._start()
+
+    def stop_online_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        self.set_mode(FapdMode.ONLINE)
+        self._stop()
+
+    def status_online_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        # ToDo: Consider returning a tuple with status and instance info
+        self.set_mode(FapdMode.ONLINE)
+        return self._status()
+
+    def start_profiling_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        self.set_mode(FapdMode.PROFILING)
+        self._start()
+
+    def stop_profiling_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        self.set_mode(FapdMode.PROFILING)
+        self._stop()
+
+    def status_profiling_fapd(self):
+        # ToDo: Add logic to that online and profiling instances are mutex
+        # ToDo: Consider returning a tuple with status and instance info
+        self.set_mode(FapdMode.PROFILING)
+        return self._status()
+
+    def set_profiling_stdout(self, strStdoutPath):
+        logging.debug("FapdManager::set_profiling_stdout()")
+        self.fapd_profiling_stdout = strStdoutPath
+
+    def set_profiling_stderr(self, strStderrPath):
+        logging.debug("FapdManager::set_profiling_stderr()")
+        self.fapd_profiling_stderr = strStderrPath
+
+    def get_profiling_stdout(self):
+        logging.debug("FapdManager::get_profiling_stdout()")
+        return self.fapd_profiling_stdout
+
+    def get_profiling_stderr(self):
+        logging.debug("FapdManager::get_profiling_stderr()")
+        return self.fapd_profiling_stderr
+
+    def set_mode(self, eMode):
+        """Allow only su to modify the instance mode. This function will be
+        removed from the public API, when integration with the Profiling
+        Tool tab is completed."""
+        self.mode = eMode
+
+    def get_mode(self):
+        """This function will be removed from the public API, when integration
+        with the Profiling Tool tab is completed."""
+        logging.debug("FapdManager::get_mode()")
+        return self.mode
+
+    def get_profiling_timestamp(self):
+        logging.debug("FapdManager::get_profiling_timestamp()")
+        return self._fapd_profiling_timestamp
```

Comment:
Consider using the `@properties` decorator for these getters/setters https://docs.python.org/3.6/library/functions.html#property

Reply:
Removed all accessor functions. Went with the "Pythonic" approach of not using accessors w/simple attributes.